### ISSUE TITLE
[WhatsApp] Can not receive messages

### ISSUE THREAD

**Issue by functored** [loqui/im#770]
Hi guys,

I'm using version 0.4.3a on FOS 2.0 . I installed it yesterday and everything appeared to work properly until I wanted to start a conversation. Apparently, I can send messages and my friends can receive them (Whatsapp on Android) but I cannot see their messages. I only receive images but only after 10 or 15 minutes. 

I saw that you guys had a similar problem last year and apparently it got solved, but now I'm wondering if I'm missing something...

Thank you!

**Comment by functored** [comment]
UPDATE: I can receive messages now but only if we are in a group. So this seems to be the same issue of last year (encryption, right?)... which means that I was confused about the availability of a solution. Anyway, please let me know your comments and/or consider closing this issue if it had been raised before.

Regards

**Comment by LOLLYTHECAT** [comment]
Same problem in OpenWapp (issue 157) ... tested with FFOS 2.0 and 2.5

**Comment by cmeerw** [comment]
I can definitely receive WhatsApp messages from Android phones with loqui.

Some log output would, of course, be helpful - either from the Firefox WebIDE or adb logcat. BTW, what I find quite useful is to have "logcat -v time -b main | grep "/Loqui IM(" | gzip -c >/storage/sdcard/loqui.log.gz &" running on the phone in the background (started in an adb shell) - that way you get the loqui console output (with some delay due to buffering) on the phone even if its not connected via a USB cable to the PC. You can then look for a specific timestamp in that file or look for something like "<message"...

**Comment by functored** [comment]
Hi!
Thanks for your reply. I'm very new to mobile programming, so I'm not very sure what should I look for in the debugging output. I'm using WebIDE and I made a couple of tests with my girlfriend. 

When she sends me messages, this is what I get in the console:

Received socket data: 256 bytes! ~ BinaryReader.prototype.onSocketData@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:5179:5 [BinaryReader]

message from="phonenumber@s.whatsapp.net" ....>
enc [object Object]="1" type="pkmsg">
XXXXX </enc
</message
 ~ BinaryReader.prototype._readNextTree@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:5279:7 [BinaryReader]

Here, I slighlty changed the contents of the tags to put it here. The XXXXX replaces what it seems an encrypted message.

Then, there are some lines about Parsing message and the BynaryWriter which seem to be nominal, followed by this:

stream:error
<ack>
</ack>
/stream:error
 ~ BinaryReader.prototype._readNextTree@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:5279:7 [BinaryReader] coseme.js:3541
null ~ BinaryReader.prototype._readNextTree@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:5279:7 [BinaryReader] coseme.js:3541
Error writing! {} ~ self._onErrorSendDisconnected@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:7580:9 [ConnectionManager] coseme.js:3541
auth_login called for 4915231779867 ~ methodList.auth_login@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:7873:9 [ConnectionManager] coseme.js:3541
Connecting to: c2.whatsapp.net:5222 ~ openConnection@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:6163:5 [auth] coseme.js:3541
Starting login... ~ tryLogin@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:6179:5 [auth] coseme.js:3541
Sending stream start, features & authentication ~ tryLogin@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:6184:5 [auth] coseme.js:3541
There is a existing challenge, trying one-shot auth! ~ getAuth@app://{61a4d487-5fac-4b42-bfe2-b0ef0953f747}/scripts/mozillahispano/coseme.js:6354:7 [auth] coseme.js:3541

... and goes on.

So, I'm not sure if this is useful. Please let me know if you need me to check at some specific messages

**Comment by cmeerw** [comment]
Ok, looks like the WhatsApp server thinks that your phone supports encryption... Did you use your phone number with an Android WhatsApp client in the past?

Not sure what to suggest - I don't have the test environment (Android phone) to reproduce or play around with it...

**Comment by bobcanthelpyou** [comment]
@cmeerw You could use Android SDK and WhatsApp Messenger APK as test environment and a second prepaid SIM card

**Comment by functored** [comment]
Ok, it seems that this is problem. Effectively, my old phone was an Android and I have not changed the number.

**Comment by cmeerw** [comment]
Ok, managed to get yowsup with encryption working now and I can at least reproduce the problem locally... Not sure how to solve it, though.

**Comment by LOLLYTHECAT** [comment]
Same problem here... and i used a android phone before using openwapp...

**Comment by LOLLYTHECAT** [comment]
news? @cmeerw

**Comment by cmeerw** [comment]
I even tried deleting the WhatsApp account from the WhatsApp server and then reactivating it and that didn't help either (although I am not sure if I should have waited a bit longer before reactivating the account)...

**Comment by LOLLYTHECAT** [comment]
@cmeerw  i can receive messages from Nokia Lumia (Microsoft)

**Comment by LOLLYTHECAT** [comment]
ok no, sorry i can receive only photos from all the devices

**Comment by pravi** [comment]
I think its not a whatsapp problem, I can't receive any message in xmpp as well. Sending works though.

0.4.4a on android

**Comment by cmeerw** [comment]
@pravi This issue is specifically for encrypted WhatsApp messages which are not yet supported by loqui. For anything else we need a separate issue with debug logs.

**Comment by FabianOvrWrt** [comment]
It started happening to me today. I can't receive messages from Android and Windows Phone.

**Comment by cmeerw** [comment]
@FabianOvrWrt have you ever used your phone number with an official WhatsApp client? Do you have any logs?

**Comment by FabianOvrWrt** [comment]
I've used it with android, but that was a few months ago. I don't have any logs right now.

**Comment by LOLLYTHECAT** [comment]
No news? i still have the problem and that is really annoying.. I think is the principal bug of the client for news users thst join FirefoxOS

**Comment by cmeerw** [comment]
BTW, does anyone have a Symbian smartphone - one that supports "WhatsApp for Nokia"?

Loqui is identifying itself as an S40, but WhatsApp on the S40 now also supports encryption. I believe the Symbian WhatsApp client is the only one left that doesn't support encryption, so maybe we could try to figure out how to identify ourselves as a Symbian phone...

**Comment by cmeerw** [comment]
Could someone with an Android phone (or from any other official WhatsApp client) try to send a WhatsApp message to +447961596923 please.

**Comment by cmeerw** [comment]
some good news, see mozillahispano/openwapp#171

**Comment by cmeerw** [comment]
Can someone test if 46f234a99e4340fc6a62b07ea490113d21fc7dd7 fixes this?

**Comment by cmeerw** [comment]
Has anyone been able to test if the recent changes fix this issue?

**Comment by xylocli** [comment]
For me it works, can you push the changes to the marketplace?

**Comment by holztest08** [comment]
It works, I can receive messages from single users again.

**Comment by cmeerw** [comment]
I have uploaded 0.4.10a with that fix to the marketplace (awaiting review)...

**Comment by cmeerw** [comment]
0.4.10a is now available from the marketplace, closing...
